generator/main.ts

```diff
@@ -55,10 +64,16 @@ async function main([out_dir, ...idl_files]) {
       return index == 0 ? match.toUpperCase() : match;
     });
   });
-
+  let interface_names = new Array<String>();
   for (let idl_file of idl_files) {
-    await generateInterface(env, idl_file, out_dir);
+    const parsedData = webidl.parse(await file.read(path.resolve(idl_file)));
+    const idl_fragments: idls.Fragments = new idls.Fragments(parsedData);
+    await generateInterface(env, idl_file, out_dir, idl_fragments);
+    if (idl_fragments.definitions[0].type == idls.DefinitionType['Interface'])
```

Comment:
i think `if (idl_fragments.definitions[0] instanceof idls.InterfaceImpl)` is more clearly.